### PR TITLE
Implement menu search and add location information

### PR DESCRIPTION
**Pull request by benjamin-shen**
### Summary 

This pull request adds menu search functionality to the Flux backend.

- [x] generalized function for creating redis cache key, refactored all routes that had queries
- [x] changed `date` query to `startDate` and `endDate` and implemented using Moment.js
- [x] implemented menu item search in `/menuData` route using String.includes() and [string-similarity](https://www.npmjs.com/package/string-similarity).compareTwoStrings(). 
- [x] added location information to the `/menuData` endpoint (related to [flux-functions PR 13](https://github.com/cornell-dti/flux-functions/pull/13))

### Test Plan 

Test by commenting out auth, running `npm run dev`, and GET requesting from `http://localhost:3333/v1/menuData?facility=<id>&startDate=<iso-date>&endDate=<iso-date>&q=<menuQuery>`. All of the queries are optional but the frontend should maintain that order for caching purposes.

The result has filtered weeksMenus/menus lists and only returns menus with items that match the query. It also returns a similarity score in range (0,1] for each menu. The frontend can then sort by this value when displaying search results.
image

### Notes 

- The menu item search is pretty naive; it may be worth looking into other methods like [Fuse.js](https://fusejs.io/examples.html).

### CODE REVIEW

**Comment by kungpaogao** [comment]
Is it possible to have the `date` query be a date range query with like `startDate` and `endDate`?

File: src/v1/server.ts [hunk 1]
```diff
@@ -7,6 +7,24 @@ import diningRoutes from './dining/routes';
...
+    return {
+      key: q,
+      value: req.query[q]
+    };
+  });
+  const qs = query
+    .filter(q => !!q.value)
+    .map(({ key, value }) => `${key}=${value}`)
+    .join('&');
+  return route + (qs && `?${  qs}`);
```
Comment (kungpaogao) [inline src/v1/server.ts]: weird space hehe

**Review by kungpaogao — DISMISSED** [review]
Looks good to me!!!!!!!!!11!!

File: src/v1/server.ts (hunk 1 above)
Comment (benjamin-shen) [reply src/v1/server.ts]: oh you're right

**Review by kungpaogao — APPROVED** [review]
no space, very good

**Pull request merged**